In [1]:
import pandas as pd

In [2]:
df1 = pd.read_csv('LaptopPrices_StarTech.csv')
df2 = pd.read_csv('LaptopPrices_Ryans.csv')

In [3]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 272 entries, 0 to 271
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Product Name    272 non-null    object
 1   Processor Type  272 non-null    object
 2   RAM             272 non-null    object
 3   Storage         272 non-null    object
 4   Display         272 non-null    object
 5   Price           272 non-null    object
dtypes: object(6)
memory usage: 12.9+ KB


In [9]:
df1.head()

Product Name  \
0  Lenovo IdeaPad D330 10IGL Celeron N4020 10.1" ...   
1  Lenovo IdeaPad 1 14AMN7 AMD Ryzen 5 14" FHD La...   
2  MSI Modern 14 C12M Core i3 12th Gen 14" FHD La...   
3  Acer Extensa 14 EX214-53-543W Core i5 12th Gen...   
4  Apple MacBook Pro 14 inch M3 Chip Liquid Retin...   

                                      Processor Type                 RAM  \
0  Intel Celeron N4020 (4M Cache,1.10 GHz up to 2...           4GB DDR4    
1          AMD Ryzen 5 7520U (2.8 GHz up to 4.3 GHz)  8GB DDR5 5500MHz,    
2  Intel Core i3-1215U (10M Cache, 3.30 GHz, up t...       8GB 3200MHz,    
3  Intel Core i5-1240P Processor (12M Cache, up t...          8GB DDR4,    
4  Apple M3 chip with 8-core CPU, 10-core GPU, 16...               8GB,    

           Storage                                    Display    Price  
0   128GB eMMC SSD          10.1" (1280 x 800) HD Touchscreen   34,500  
1       256GB  SSD                        14" FHD (1920X1080)   60,500  
2        512GB SSD                      14" FHD (1920 x 1080)   59,999  
3        512GB SSD                      14" FHD (1920 x 1080)   79,999  
4        512GB SSD  14.2" Liquid Retina display (3024 x 1664)  228,000

In [4]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 349 entries, 0 to 348
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Product Name    349 non-null    object 
 1   Processor Type  349 non-null    object 
 2   RAM             349 non-null    object 
 3   Storage         349 non-null    object 
 4   Display         349 non-null    float64
 5   Price           349 non-null    object 
dtypes: float64(1), object(5)
memory usage: 16.5+ KB


In [10]:
df2.head()

Product Name Processor Type  RAM  \
0  Lenovo IdeaPad D330 10IGL Intel CDC N4020 4GB ...  Intel Celeron  4GB   
1  HP 15s-eq1578AU AMD Athlon Silver 3050U 8GB  2...     AMD Athlon  8GB   
2  Acer Aspire 3 A315-510P-38RH (13th Gen Standar...        Core i3  8GB   
3  Lenovo IdeaPad 1 15ADA7 AMD Ryzen 3 3250U 8GB ...        Ryzen 3  8GB   
4  Lenovo IdeaPad Slim 3i 15ITL 11th Gen Intel Co...        Core i3  4GB   

      Storage  Display           Price  
0  128GB eMMC     10.1         34,500   
1   256GB SSD     15.6  46,500 48,900   
2   512GB SSD     15.6  49,800 53,000   
3   512GB SSD     15.6         49,900   
4   256GB SSD     15.6         51,400

In [5]:
df1.shape

(272, 6)

In [6]:
df2.shape

(349, 6)

In [7]:
duplicates = df1[df1['Product Name'].duplicated(keep=False)]
num_duplicates = len(duplicates)
print(f'Number of duplicate products: {num_duplicates}')

Number of duplicate products: 0


In [8]:
duplicates_df2 = df2[df2['Product Name'].duplicated(keep=False)]
num_duplicates_df2 = len(duplicates_df2)
print(f'Number of duplicate products: {num_duplicates_df2}')

Number of duplicate products: 0


In [14]:
LaptopPrices = pd.concat([df1, df2], ignore_index=True)
unique_products = LaptopPrices['Product Name'].unique()
num_unique_products = len(unique_products)
print(f'Number of unique products: {num_unique_products}')

Number of unique products: 621


In [15]:
LaptopPrices.head()

Product Name  \
0  Lenovo IdeaPad D330 10IGL Celeron N4020 10.1" ...   
1  Lenovo IdeaPad 1 14AMN7 AMD Ryzen 5 14" FHD La...   
2  MSI Modern 14 C12M Core i3 12th Gen 14" FHD La...   
3  Acer Extensa 14 EX214-53-543W Core i5 12th Gen...   
4  Apple MacBook Pro 14 inch M3 Chip Liquid Retin...   

                                      Processor Type                 RAM  \
0  Intel Celeron N4020 (4M Cache,1.10 GHz up to 2...           4GB DDR4    
1          AMD Ryzen 5 7520U (2.8 GHz up to 4.3 GHz)  8GB DDR5 5500MHz,    
2  Intel Core i3-1215U (10M Cache, 3.30 GHz, up t...       8GB 3200MHz,    
3  Intel Core i5-1240P Processor (12M Cache, up t...          8GB DDR4,    
4  Apple M3 chip with 8-core CPU, 10-core GPU, 16...               8GB,    

           Storage                                    Display    Price  
0   128GB eMMC SSD          10.1" (1280 x 800) HD Touchscreen   34,500  
1       256GB  SSD                        14" FHD (1920X1080)   60,500  
2        512GB SSD                      14" FHD (1920 x 1080)   59,999  
3        512GB SSD                      14" FHD (1920 x 1080)   79,999  
4        512GB SSD  14.2" Liquid Retina display (3024 x 1664)  228,000

In [16]:
LaptopPrices

Product Name  \
0    Lenovo IdeaPad D330 10IGL Celeron N4020 10.1" ...   
1    Lenovo IdeaPad 1 14AMN7 AMD Ryzen 5 14" FHD La...   
2    MSI Modern 14 C12M Core i3 12th Gen 14" FHD La...   
3    Acer Extensa 14 EX214-53-543W Core i5 12th Gen...   
4    Apple MacBook Pro 14 inch M3 Chip Liquid Retin...   
..                                                 ...   
616  Apple MacBook Pro (2023) Apple M2 Max Chip 32G...   
617  Apple MacBook Pro (Late 2021) Apple M1 Max Chi...   
618  Apple Macbook Pro (Early 2023) Apple M2 Max Ch...   
619  Apple Macbook Pro (Late 2023) Apple M3 Max 48G...   
620  Apple Macbook Pro (Late 2023) Apple M3 Max 48G...   

                                        Processor Type                 RAM  \
0    Intel Celeron N4020 (4M Cache,1.10 GHz up to 2...           4GB DDR4    
1            AMD Ryzen 5 7520U (2.8 GHz up to 4.3 GHz)  8GB DDR5 5500MHz,    
2    Intel Core i3-1215U (10M Cache, 3.30 GHz, up t...       8GB 3200MHz,    
3    Intel Core i5-1240P Processor (12M Cache, up t...          8GB DDR4,    
4    Apple M3 chip with 8-core CPU, 10-core GPU, 16...               8GB,    
..                                                 ...                 ...   
616                                           Apple M2                32GB   
617                                           Apple M3                64GB   
618                                           Apple M3                32GB   
619                                           Apple M3                48GB   
620                                           Apple M3                48GB   

             Storage                                    Display  \
0     128GB eMMC SSD          10.1" (1280 x 800) HD Touchscreen   
1         256GB  SSD                        14" FHD (1920X1080)   
2          512GB SSD                      14" FHD (1920 x 1080)   
3          512GB SSD                      14" FHD (1920 x 1080)   
4          512GB SSD  14.2" Liquid Retina display (3024 x 1664)   
..               ...                                        ...   
616          1TB SSD                                       16.2   
617          4TB SSD                                       16.2   
618          2TB SSD                                       16.2   
619          1TB SSD                                       16.2   
620          1TB SSD                                       16.2   

                   Price  
0                 34,500  
1                 60,500  
2                 59,999  
3                 79,999  
4                228,000  
..                   ...  
616             465,000   
617             479,000   
618  495,000 (Estimated)  
619  538,000 (Estimated)  
620  540,000 (Estimated)  

[621 rows x 6 columns]

In [ ]:
LaptopPrices.to_csv('D:\GitHub Repos\Laptop-Prices-Web-Scraping/LaptopPrices.csv')

**A combined dataset namely "LaptopPrices" has been created with the data scraped off two websites (Ryans Computers and Star Tech).** 
The dataset has has 621 observations/ rows and 6 columns ["Product Name", "Processor Type", "RAM", "Storage", "Display", "Price"]